osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -31,6 +33,8 @@ public class BeatmapCarousel : CompositeDrawable, IKeyBindingHandler<GlobalActio
         private const float bleed_top = FilterControl.HEIGHT;
         private const float bleed_bottom = Footer.HEIGHT;
 
+        private readonly Bindable<User> localUser = new Bindable<User>();
```

Comment:
I have mostly negative feelings about the carousel being API- and `User`-dependent with this change. I'd say suggestion logic should be hidden somewhere so that the carousel logic (which is already convoluted in places) doesn't now become entirely unreadable and untestable offline due to online coupling.